src/main/java/com/google/cloud/spanner/pgadapter/parsers/ArrayParser.java

```diff
@@ -70,9 +70,8 @@ private List<?> toList(Value value, Code arrayElementType) {
       case INT64:
         return value.getInt64Array();
       case STRING:
-        return value.getStringArray();
       case PG_NUMERIC:
-        return value.getNumericArray();
+        return value.getStringArray();
```

Comment:
Not able to understand this.

Reply:
ah, good point. I've added the following comment to the code to clarify:

        // Get numeric arrays as a string array instead of as an array of BigDecimal, as numeric
        // arrays could contain 'NaN' values, which are not supported by BigDecimal.